### ISSUE TITLE
media(imags,videos) management page

### ISSUE THREAD

**Issue by nextfool** [nopSolutions/nopCommerce#5710]
Can we have a media/image management page just like other entities(product, category ...etc), so that we can:

1.Create(upload) new image, not for product/category pages use, but for other pages(about us, contact us, or even refer image in email templates). This is useful because we want to take advantage of Nop's built in resize function.
2.Quick change any images' attributes or even change the whole image without changing its id.
3.A central place to manage images, currently I personally feel some uncertainty because Nop controls everything internally.

As the above No.1 says, the biggest problem with current implementation is that, all images are only for product/category use, and you can only upload in their create/edit page. We would like to upload some common images for other places to refer and still be able to resize with Nop's image resize system.

Can this be done?

**Comment by nextfool** [comment]
Or at least create an api/endpoint, so that we can pass the image path, width/height and other parameters, to get the resized image, of cause the result image is cached in file system so this api doesn't have to run each time.

Basically something like this:

http//domain.com/image-resize?path=image-relative-path&width=480&any-other-parameters.

**Comment by untiedshoes** [comment]
Something like ‘media’ in umbraco would be a great addition.

**Comment by untiedshoes** [comment]
> Basically something like this:
> 
> http//domain.com/image-resize?path=image-relative-path&width=480&any-other-parameters.

A few years ago, with a 3.00 & 3.90 site, I did this using ImageProcessor & ImageProcesor.Web, but once we moved that site to using a CDN for images, it didn't work, as the images we're resized on the fly. Umbraco (at the time), had an ImageProcessor plugin, that resized on the fly, but also uploaded that image to Azure too, but I couldn't get it to work.

It also enabled me to convert to WEBP & PNG on the fly by doing this.

`@(Model.DefaultPictureModel.FullSizeImageUrl)?quality=95&width=300&format=webp`
